src/components/encounter/encounter-form-manager.ts

```diff
@@ -174,3 +234,23 @@ function hasSubmittableObs(field: FormField) {
   }
   return !field.isHidden && !field.isParentHidden && (type === 'obsGroup' || hasSubmission(field));
 }
+
+export function getPatientProgram(patientPrograms: Array<PatientProgram>, programUuid: string) {
+  const programs = patientPrograms.find((program) => program.program.uuid === programUuid);
+  return programs;
+}
+
+function retirePreviousState(currentStates, previousStates) {
+  previousStates?.forEach((previousState) => {
+    const matchingObj = currentStates?.find((currentState) => currentState?.state == previousState?.state.uuid);
+    if (!matchingObj) {
+      const formattedMatchedObj = {
+        state: previousState?.state?.uuid,
+        startDate: previousState.startDate,
+        endDate: previousState.startDate,
+      };
+      currentStates.push(formattedMatchedObj);
+    }
+  });
+  return currentStates;
```

Comment:
It's little bit inorganic for a function that retires objects to return `currentObjects`, I would expect such function to return the retired objects unless it's doing more than "retiring".

Reply:
After transitioning a patient to a different state, you retire the previous state with the same uuid and pass both in the new array